lighthouse-core/scripts/i18n/collect-strings.js

```diff
@@ -30,48 +31,61 @@ const ignoredPathComponents = [
   '**/*-renderer.js',
 ];
 
-// @ts-ignore - @types/esprima lacks all of these
-function computeDescription(ast, property, value, startRange) {
-  const endRange = property.range[0];
-
-  for (const comment of ast.comments || []) {
-    if (comment.range[0] < startRange) continue;
-    if (comment.range[0] > endRange) continue;
-    if (comment.value.includes('@')) {
-      // This is a complex description with description and examples.
-      let description = '';
-      /** @type {Record<string, string>} */
-      const examples = {};
-
-      const r = /@(\w+) ({.+})?(.*)(\n|$)/g;
-      let matches;
-      while ((matches = r.exec(comment.value)) !== null) {
-        const tagName = matches[1];
-        const example = matches[2];
-        const messageOrPlaceholder = matches[3].trim();
-
-        if (tagName === 'description') {
-          description = messageOrPlaceholder;
-        } else if (tagName === 'example') {
-          examples[messageOrPlaceholder] = example.substring(1, example.length - 1);
-        }
+/**
+ * Extract the description and examples (if any) from a jsDoc annotation.
+ * @param {import('typescript').JSDoc} ast
+ * @param {string} message
+ * @return {{description: string, examples: Record<string, string>}}
+ */
+function computeDescription(ast, message) {
+  if (ast.tags) {
+    // This is a complex description with description and examples.
+    let description = '';
+    /** @type {Record<string, string>} */
+    const examples = {};
+
+    for (const tag of ast.tags) {
+      const comment = coerceToSingleLineAndTrim(tag.comment);
+
+      if (tag.tagName.text === 'description') {
+        description = comment;
+      } else if (tag.tagName.text === 'example') {
+        const {placeholderName, exampleValue} = parseAtExample(comment);
+        examples[placeholderName] = exampleValue;
       }
-
-      // Make sure description is not empty
-      if (description.length === 0) throw Error(`Empty @description for message "${value}"`);
-      return {description, examples};
     }
 
-    /** @type {string} */
-    const description = comment.value.replace('*', '').trim();
-
-    // Make sure description is not empty
-    if (description.length === 0) throw Error(`Empty description for message "${value}"`);
+    if (description.length === 0) throw Error(`Empty @description for message "${message}"`);
+    return {description, examples};
+  }
 
+  if (ast.comment) {
     // The entire comment is the description, so return everything.
-    return {description};
+    return {description: coerceToSingleLineAndTrim(ast.comment), examples: {}};
   }
-  throw Error(`No Description for message "${value}"`);
+
+  throw Error(`Missing description comment for message "${message}"`);
+}
+
+/**
+ * Line breaks within a jsdoc comment should always be replaceable with a space.
```

Comment:
:heart_eyes: getting multi line for free

I feel like this function description is worded more like a rule of the code, and not what the function does. I.e
```suggestion
Collapses a jsdoc comment into a single line and trims whitespace.
```

Reply:
> I feel like this function description is worded more like a rule of the code, and not what the function does

fair, fixed